src/hooks/useNewLiquidityMiningCampaign.ts

```diff
@@ -7,47 +16,108 @@ import { getLpTokenPrice } from '../utils/prices'
 import { useNativeCurrency } from './useNativeCurrency'
 import { usePairReserveNativeCurrency } from './usePairReserveNativeCurrency'
 
+import { useTokenDerivedNativeCurrency } from './useTokenDerivedNativeCurrency'
+import { parseUnits } from 'arb-ts/node_modules/@ethersproject/units'
+
+import { getAddress } from 'ethers/lib/utils'
+import Decimal from 'decimal.js-light'
+
 export function useNewLiquidityMiningCampaign(
-  targetedPair: Pair | null,
+  targetedPairOrToken: Pair | Token | null,
   rewards: TokenAmount[],
   startTime: Date | null,
   endTime: Date | null,
   locked: boolean,
-  stakingCap: TokenAmount | null
-): LiquidityMiningCampaign | null {
+  stakingCap: TokenAmount | null,
+  simulatedStakedAmount: string
+): LiquidityMiningCampaign | SingleSidedLiquidityMiningCampaign | null {
   const { chainId } = useActiveWeb3React()
+
+  const tokenDerivedNative = useTokenDerivedNativeCurrency(
+    targetedPairOrToken instanceof Token ? targetedPairOrToken : undefined
+  )
+
   const nativeCurrency = useNativeCurrency()
   const { pricedTokenAmounts: pricedRewardAmounts } = useNativeCurrencyPricedTokenAmounts(rewards)
-  const lpTokenTotalSupply = usePairLiquidityTokenTotalSupply(targetedPair || undefined)
-  const { reserveNativeCurrency: targetedPairReserveNativeCurrency } = usePairReserveNativeCurrency()
+  const lpTokenTotalSupply = usePairLiquidityTokenTotalSupply(
+    targetedPairOrToken instanceof Pair ? targetedPairOrToken : undefined
```

Comment:
We can make use of a `isPair` and `isToken` here

Reply:
those functions are not for this use case if you look into type they accept...